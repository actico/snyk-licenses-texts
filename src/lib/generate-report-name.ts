import { OrgData } from "./get-org-data";

export function generateReportName(orgData: OrgData, view: string): string {
  return `${orgData.slug}-${orgData.id}-${view}`;
}
