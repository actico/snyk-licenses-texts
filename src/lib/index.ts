import 'source-map-support/register';
import * as fetch from 'node-fetch';
import * as debugLib from 'debug';
import * as cheerio from 'cheerio';

export * from './license-text';
