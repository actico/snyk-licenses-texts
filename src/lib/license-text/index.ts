export { fetchNonSpdxLicenseText } from './non-spdx';
export { fetchSpdxLicenseText } from './spdx';
