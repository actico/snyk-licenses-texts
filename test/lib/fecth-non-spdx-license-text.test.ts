import { fetchNonSpdxLicenseText } from '../../src/lib';

test('ASPSecurityKit-Khosla-Tech license fetched locally as expected', async () => {
  const licenseText = await fetchNonSpdxLicenseText('ASPSecurityKit-Khosla-Tech');
  console.log(licenseText)
  expect(licenseText).toMatchSnapshot();
});

test('Failed to fetch a license returns helper text instead', async () => {
  const licenseText = await fetchNonSpdxLicenseText('AFL-1');
  expect(licenseText).toMatchSnapshot();
});
