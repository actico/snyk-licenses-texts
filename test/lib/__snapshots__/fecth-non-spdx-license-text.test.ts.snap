// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ASPSecurityKit-Khosla-Tech license fetched locally as expected 1`] = `
"<h1>KHOSLA TECH - END USER AGREEMENT</h1>
<p><strong>IMPORTANT – PLEASE READ THIS AGREEMENT!</strong></p>
<div>
    <ul>
        <li>This End User Agreement (\\"Agreement\\") is a binding legal document between Khosla Tech and you, which explains your rights and obligations as an End User of Khosla Tech products. \\"End User\\" means either (a) you as an individual or (b) your company, if you are using Khosla Tech products in your capacity as an employee or agent of a company. You identify the specific \\"End User\\" during the registration process. \\"Khosla Tech\\" means Khosla Tech Private Limited.</li>
        <li>By allowing the \\"I Agree to the terms and conditions of the End User Agreement\\" box to remain checked, or by installing or using any Khosla Tech products, End User agrees to be bound by this Agreement. If you do not agree to this Agreement, then uncheck the \\"I Accept\\" box and do not register, install or use Khosla Tech products.</li>
        <li>The \\"Agreement\\" also includes any Khosla Tech policies or documents referenced in this document, including Khosla Tech's <a href=\\"/privacy\\">Privacy Policy </a>.</li>
        <li>From time to time, Khosla Tech may modify this Agreement, including any referenced policies and other documents. Any modified version will be effective at the time it is posted. To keep abreast of your license rights and relevant restrictions, please bookmark this Agreement and read it periodically. By using any Product after any modifications, End User agrees to all of the modifications.</li>
    </ul>
    <div></div>
    <h2>1. Introduction to Khosla Tech's Products and License Terms.</h2>
    <p><strong>1.1 What's Included in \\"Software\\".</strong> Khosla Tech offers many different software products for download, including ASP Security Kit and others. \\"Software\\" means any Khosla Tech product that End User orders from Khosla Tech and downloads from Khosla Tech or from third-party sites such as nuget.org. The term also includes (a) related documentation, (b) any Updates (defined below) to the Software, (c) any add-ons, plug-ins, APIs or Internet-based components of the Software provided by Khosla Tech (but not third parties) (\\"Supplementary Software\\"), and (d) any third party software embedded in or provided with Khosla Tech's software (\\"Embedded Software\\"). End User's detailed rights to use Software are in Section 2 below.</p>
    <p><strong>1.2 The \\" Hosted Services\\".</strong> This Agreement uses \\"Hosted Services\\" to refer to both Hosted Services and APIs provided as part of various Khosla Tech’s product offerings.</p>
    <p><strong>1.3 The \\"Products\\".</strong> This Agreement uses \\"Products\\" to refer to the Software, websites, Hosted Services and APIs owned and operated by Khosla Tech.</p>
    <p><strong>1.4 Limits on Authorized Users.</strong> When buying a license or a subscription to the products, End User pays fees for a specific number of users, which are counted based on number of then-authorized users for Software and number of seats or logins (\\"Authorized Users\\"). End User designates Authorized Users through the Products. Authorized Users may include End User's employees, representatives, consultants, and contractors. End User may add Authorized Users for additional fees. End User is responsible for all use of the Products by its Authorized Users and their compliance with this Agreement.</p>
    <p><strong>1.5 Team License.</strong> When buying a license or a subscription to the products under a team license, User pays fees for a specific number and a specific kind of users. A team is defined as a group of users who work together in a company on a specific project at a time and may include employees, consultants and contractors. Team license/subscription does not cover all the employees in a company – a company may have many teams; team license is applicable only to one of the teams for whom the licensed is originally purchased. Khosla Tech reserves the right to change the number of authorized users allowed for a team license at any time and may charge End User additional fee for each additional user based on the prevailing rate at that time.</p>
    <p><strong>1.6 Support.</strong> Unless otherwise specified, consultancy and phone support would be provided using Skype. Consultancy hours are provided to help you integrate and use ASP Security Kit effectively in your project and not for other purposes such as fixing technical issues in your system or general help related to building sites on ASP.NET ETC.</p>
    <div></div>
    <h2>2. License to Use Khosla Tech Software.</h2>
    <p><strong>2.1 Software, Generally.</strong> Khosla Tech grants End User a worldwide, non-exclusive, non-transferable, non-sublicenseable right to use the Software, subject to the terms and conditions of this Agreement. Software licenses are perpetual unless terminated as described in this Agreement. Only Authorized Users may use the Software, and only up to the permitted number of concurrent Authorized Users (except for Non-Production Purposes, as described below). All use of Software must be in accordance with the relevant Khosla Tech documentation and policies. End User may make a limited number of copies of the Software as is strictly necessary for purposes of data protection, archiving, backup, and testing and internal development (e.g., of End User Modifications or End-User Plug-ins, as defined below, but not development of End User's unrelated products or services) (\\"Non-Production Purposes\\"). For clarity, End User's employees and contractors may use the Software for these Non-Production Purposes, and End User will be responsible for their compliance with this Agreement on all the same terms as for Authorized Users, except that these users will not count against End User's limit of Authorized Users, and End User will not need to pay additional fees for these users. Unless Khosla Tech agrees otherwise in writing, End User may only install, use or make available the Software on End User's hardware systems, whether owned, leased or controlled. End User will be responsible for any use of the Software on any hardware systems not owned, leased or controlled by End User (\\"Uncontrolled Systems\\").</p>
    <p><strong>2.2 Updates and Supplementary Software.</strong> The terms for Software in this Agreement apply to any Updates and Supplementary Software, unless Khosla Tech provides different terms. Khosla Tech may cease making available Supplementary Software and Internet-based services used with the Software (e.g., the Khosla Tech APIs) at any time.</p>
    <p><strong>2.3 Open Source Software.</strong> The Software may contain or be provided with components subject to the terms and conditions of \\"open source\\" software licenses (\\"Open Source Software\\"). To the extent required by the license that accompanies the Open Source Software, the terms of such license will apply in lieu of the terms of this Agreement with respect to such Open Source Software, including any provisions governing access to source code, modification or reverse engineering.</p>
    <p><strong>2.4 Evaluation Software.</strong> This paragraph applies to any Software that Khosla Tech makes available on an evaluation basis (\\"Evaluation Software\\"). End User may only use the Evaluation Software for internal evaluation purposes for the period specified by Khosla Tech (or, if not specified, for 30 days), and may only permit a limited number of users (specified by Khosla Tech) to access the Evaluation Software. After the evaluation period, End User must delete all copies of the Evaluation Software. End User acknowledges that Evaluation Software may not be fully functional. Notwithstanding anything else in this Agreement, Khosla Tech does not offer any warranty, indemnity or support for any Evaluation Software.</p>
    <p><strong>2.5 Source Code.</strong> Khosla Tech may provide some elements of Software in source code form (\\"Source Code\\"). Unless otherwise specified, End User may modify Source Code solely to develop bug fixes, customizations, and additional features (\\"End User Modifications\\") and, notwithstanding anything else in this Agreement, may only use End User Modifications internally for purposes of using the Software licensed from Khosla Tech. End User may not use the source code to build products or services that resemble or compete with Khosla Tech’s products and services. Khosla Tech will have no support, warranty, indemnity or other obligations relating to, and assumes no liability for, any End User Modifications or any effect they may have on the operation of the Products.</p>
    <p><strong>2.6 IP Protection Mechanisms.</strong> The Software has license protection mechanisms designed to manage and protect Khosla Tech's and its suppliers' and licensors' intellectual property rights. Whether using Source Code or not, End User must not modify or alter these mechanisms or try to circumvent them or the usage rules they are designed to enforce.</p>
    <h2>3. Use of Khosla Tech Hosted Services and APIs.</h2>
    <p><strong>3.1 Access Rights.</strong> End User may access and use the Hosted Services during the applicable Subscription Term (defined below), subject to the terms and conditions of this Agreement. Only Authorized Users may use the Hosted Services, and only up to the permitted number of Authorized Users. All use of the Hosted Services must be in accordance with the relevant Khosla Tech documentation and policies.</p>
    <p><strong>3.2 Subscription and Renewals.</strong> End User selects its initial subscription term, which may be a month, quarter, year, or other mutually agreed period, at the time of order. Once that term expires, it will automatically renew for successive terms of the same period (but no longer than 1 year each) unless either End User or Khosla Tech notifies the other of non-renewal at least 5 business days prior to the upcoming expiration date or Khosla Tech ceases to make a particular Hosted Service available. Renewals are charged at Khosla Tech's then-current rates unless otherwise agreed, and Khosla Tech will at its discretion charge End User using the credit card on file on or after the expiration date or send End User an invoice. \\"Subscription Term\\" means the initial term and any renewal term(s).</p>
    <p><strong>3.3 Evaluations.</strong> For evaluations of Hosted Services, the \\"Subscription Term\\" does not apply. Instead, the evaluation period is specified by Khosla Tech. Khosla Tech may notify End User of the remaining number of days through the Hosted Services. Notwithstanding anything else in this Agreement, Khosla Tech does not offer any warranty, indemnity or support for any Hosted Services offered on an evaluation basis.</p>
    <p><strong>3.4 End User Data.</strong> any data, content or information that End User provides through the Hosted Services must comply with all domestic, foreign and international laws and regulations, including those relating to data privacy, international communications, and the exportation of technical or personal data. End User is solely responsible for the accuracy, content and legality of all End User Data. End User agrees not to submit any content that is obscene, defamatory, libelous, threatening, harassing, pornographic, racially or ethnically offensive, that encourages conduct that would be considered a criminal offense or give rise to any civil liability. End User will not submit any material (including any virus, bot, worm, scripting exploit or other harmful code) that is likely to harm or corrupt the Hosted Services or any computer systems or data. End User represents and warrants to Khosla Tech that it has sufficient rights in the End User Data to grant the rights in this section and that the End User Data does not infringe the rights of any third party. End User agrees that Khosla Tech may, in its sole discretion, delete or remove any End User Data at any time and with or without notice.</p>
    <p>End User hereby grants Khosla Tech a non-exclusive license to copy, distribute, perform, display, store, modify, and otherwise use End User Data in connection with operating the Hosted Services.</p>
    <p>Khosla Tech may also use aggregate information to measure general Hosted Service usage patterns and characteristics of its user base and otherwise to improve its products and services, and may include such aggregate information about its audience in promotional materials or reports to third parties. This aggregate information will not reference names, phone numbers, email addresses, or other personally identifiable information, and it will be not traceable to a specific recipient or user email address.</p>
    <p><strong>3.6 Return of End User Data.</strong> After termination or expiration of a Subscription Term or this Agreement, unless it was for End User's breach, at End User's request, Khosla Tech will use reasonable efforts to make available the End User Data for download. If made available, End User must download the End User Data within 30 days of termination or expiration. After that, Khosla Tech may delete the End User Data.</p>
    <p><strong>3.5 End User Responsibility for Systems.</strong> End User is responsible for its own Internet connection and must use software, systems and equipment compatible with the Hosted Services, as Khosla Tech specifies in its published policies. Any End User web browsers and other software must support the Secure Socket Layer (SSL) protocol or other protocols accepted by Khosla Tech. Khosla Tech is not responsible for any End User Data lost, altered, intercepted or stored across networks not owned or operated by Khosla Tech.</p>
    <h2>4. Important Customer Obligations</h2>
    <p><strong>4.1 Accounts and Passwords.</strong> End User will provide accurate, current and complete information when registering with Khosla Tech and ordering Products and agrees to update its information if it changes. This is important, because Khosla Tech may send notices, statements and other information to End User by email or through End User's account (for Hosted Services). End User will keep all its Authorized Users' passwords and usernames confidential and will not share them with third parties. End User is responsible for all actions taken through its accounts.</p>
    <p><strong>4.2 Embedded Software.</strong> All of the other restrictions for Software in this Agreement also apply to Embedded Software Software, with the addition of the following terms. End User receives restricted licenses to Embedded Software and may use Embedded Software only for its internal purposes (including for Customer Use as permitted below) in conjunction with the applicable Product as provided by Khosla Tech, and may only use the Embedded Software as part of and through that Product. End User may not install, access, configure or use any Embedded Software (including any APIs, tools, databases or other components of any Embedded Software) separately or independently of the rest of the Product, whether for production, technical support or any other purposes, or otherwise attempt to gain direct access to any Embedded Software components, or permit anyone else (including Customers) to do any of these things. Notwithstanding any other terms of this Agreement, End User may not modify any Embedded Software. End User will be financially responsible to the applicable third party licensor (\\"Embedded Software Licensor\\") for all damages and losses resulting from End User's or its Customers' breach of this Agreement. End User may not \\"benchmark\\" or otherwise analyze performance information for individual Embedded Software elements.</p>
    <p>Some Embedded Software may include source code provided as part of the Embedded Software Licensor's standard shipment. That source code will be governed by the terms for Embedded Software in this Agreement, and not the terms for Source Code in Section 2.5 (Source Code) above.</p>
    <p>End User understands that the applicable Embedded Software Licensor retains all ownership and intellectual property rights to the Embedded Software. Embedded Software Licensors (and any other third party licensors of any components of the Products) are intended third party beneficiaries of this Agreement with respect to the items they license and may enforce this Agreement directly against End User; but, to be clear, Embedded Software Licensor do not assume any of Khosla Tech's obligations under this Agreement. In addition, End User agrees to the Embedded Software-related audit provisions in Section 6 (License Certifications and Audits) below.</p>
    <p>Embedded Software may include or be accompanied by third party technology that may be appropriate or necessary for use with some Embedded Software. Such technology may be specified in the Documentation or otherwise specified by Khosla Tech, and is licensed to End User only for use with the Product with which it is provided under the specified third party license terms, and not this Agreement. Khosla Tech may also provide additional or substitute terms for Embedded Software if required by the Embedded Software Licensor, and End User will comply with those terms.</p>
    <p><strong>4.3 Rules for Plug-ins.</strong> End User may develop its own plug-ins for various Products (\\"End User Plug-ins\\") and may distribute End User Plug-ins to third parties, but only for those Products permitted by Khosla Tech, and only in accordance with Khosla Tech's published plug-in and API guidelines. End User Plug-ins distributed through the Khosla Tech Marketplace will be subject to the posted terms for the Khosla Tech Marketplace.</p>
    <p><strong>4.4 Conditions on Use of Products.</strong> End User will not, and will not allow any Authorized User or other third party to: (a) resell, assign, rent, give, transfer, pass title to, lease, copy, provide access to or sublicense (including without limitation on a timeshare, subscription service, hosted service or outsourced basis) any Product to any third party (for use in its business operations or otherwise) or anyone else besides Authorized Users in accordance with this express terms of this Agreement, or permit anyone besides Authorized Users to use any data or information not owned by End User that is generated by the Products (and, in the event End User grants any security interest in any Products, the secured party has no right to use or transfer the Products); (b) reverse engineer, decompile, disassemble, decipher, decrypt, or otherwise seek to discover or obtain the source code or non-public APIs to Products (including any data structure or similar materials produced by the Products), except to the extent expressly permitted by applicable law despite this prohibition (and then only upon advance notice to Khosla Tech); (c) modify, adapt or create derivative works of a Product to build products or services that resemble or compete with Khosla Tech’s products and services; (d) remove or obscure any proprietary or other notices of Khosla Tech or any third party contained in any Product (including any information or data generated by the Products); (e) publicly disseminate information regarding the performance and the security of Products; (f) use any Product for commercial solicitation purposes or spam; (g) use the Khosla Tech name or any Khosla Tech trademarks or logos without Khosla Tech's prior written consent; or (h) commit any act or omission that could result in damage to Khosla Tech's or its suppliers' or licensors' reputations. End User will not attempt to do any of the foregoing, encourage others to do so, or otherwise attempt to bypass or circumvent any usage restrictions in this Agreement.</p>
    <p><strong>4.5 Indemnification.</strong> End User will indemnify, defend and hold harmless Khosla Tech from and against any and all claims, costs, damages, losses, liabilities and expenses (including reasonable attorneys' fees and costs) arising out of or in connection with any claim arising from or relating to (a) any breach by End User (including any Authorized Users) of this Agreement, (b) any End User Data, (c) any End User Modifications, End User Plug-ins or other modifications of or combinations with a Product, or any service or product offered by End User in connection with or related to a Product, (d) any Uncontrolled Systems, or (e) any representations or warranties made by End User (including any Authorized User) regarding a Product to third parties. This indemnification obligation is subject to End User receiving (i) prompt written notice of such claim (but in any event notice in sufficient time for End User to respond without prejudice); (ii) the exclusive right to control and direct the investigation, defense, or settlement of such claim, provided that Khosla Tech may participate in the claim at its own expense and End User may not settle any claim without Khosla Tech's prior written consent; and (iii) all reasonable necessary cooperation of Khosla Tech at End User's expense.</p>

    <h2>5. Fees and Payment</h2>
    <p><strong>5.1 General.</strong> End User must pay all fees by their due date specified at the time of order or otherwise within 30 days of Khosla Tech's invoice or notice. Renewal fees for Hosted Services will be charged as described in Section 3.2 (Subscription and Renewals). Late payments are subject to interest charges of 1.0% per month on any outstanding balance, or the maximum permitted by law, whichever is less, plus all collection expenses. End User will continue to be charged during any period of suspension. In event of any termination, End User will pay the unpaid balance due calculated in accordance with this section and this Agreement. Khosla Tech may charge such unpaid fees and charges to End User's credit card or otherwise bill End User for such unpaid fees and charges. End User acknowledges that it is not relying on the future availability of any Products (including any Embedded Software) in agreeing to or making its payments hereunder.</p>
    <p><strong>5.2 Taxes.</strong> Payments made by End User under this Agreement exclude any taxes or duties payable in respect of the Products in the jurisdiction where the payment is either made or received. To the extent that any such taxes or duties are payable by Khosla Tech, End User must pay to Khosla Tech the amount of such taxes or duties in addition to any fees owed under this Agreement. Notwithstanding the foregoing, End User may have obtained an exemption from relevant taxes or duties as of the time such taxes or duties are levied or assessed. In that case, End User will have the right to provide to Khosla Tech with any such exemption information and Khosla Tech will use reasonable efforts to provide such invoicing documents as may enable End User to obtain a refund or credit for the amount so paid from any relevant revenue authority if such a refund or credit is available.</p>

    <h2>6. License Certifications and Audits</h2>
    <p>Upon Khosla Tech's written request, End User will provide Khosla Tech with a signed certification certifying that all Products are being used pursuant to the terms of this Agreement, including any access and user limitations. With prior reasonable notice of at least ten (10) days, Khosla Tech (or its authorized agent) may audit the use of the Products by End User and its Authorized Users and any Customers, provided such audit is during regular business hours. End User will provide reasonable assistance and access to information in the course of any audit. End User is responsible for such audit costs only in the event the audit reveals that End User's use of the Products is not in accordance with the permitted scope of use. In the event that any certification or audit reveals that End User has exceeded its permitted number of Authorized Users, Khosla Tech may invoice End User for any past or ongoing excessive use and End User will pay the invoice in accordance with Section 5 (Fees and Payment). This remedy is without prejudice to any other remedies available to Khosla Tech at law or equity or under this Agreement.</p>
    <p>End User understands that Khosla Tech may report audit results to any applicable Embedded Software Licensor or may assign the right to audit End User in this Section 6 to Embedded Software Licensors. When the audit is assigned, then notwithstanding anything else in this Agreement to the contrary, End User Licensor will not be responsible for End User's costs incurred in cooperating with the audit.</p>

    <h2>7. Khosla Tech's Ownership Rights</h2>
    <p>Khosla Tech and its licensors reserve all rights not expressly granted to End User in this Agreement. The Products (including any content or information contained therein) and all copies thereof are protected by copyright and other intellectual property laws and treaties. Khosla Tech or its licensors own the title, copyright, and other intellectual property rights in the Products and all copies, modifications and derivative works of the Products and underlying software (including any incorporating Feedback) (\\"Khosla Tech Technology\\"), and End User does not acquire any ownership rights in Khosla Tech Technology. All Products are licensed, not sold.</p>
    <p>\\"Feedback\\" means any feedback, comments, suggestions or materials (including, to the extent disclosed to Khosla Tech, any End User Modifications, but excluding End User Plug-ins) that End User may provide to Khosla Tech about or in connection with the Products, including any ideas, concepts, know-how or techniques contained therein. End User may provide Feedback in connection with Maintenance and otherwise. End User hereby grants Khosla Tech a worldwide, royalty-free, non-exclusive, perpetual and irrevocable license to use, copy, modify and otherwise exploit the Feedback for any purpose, including incorporating or implementing the Feedback in the Products. End User agrees that Khosla Tech may exploit all Feedback without any restriction or obligation on account of intellectual property rights or otherwise. For clarity, no Feedback will be deemed End User's Confidential Information, and nothing in this Agreement (including Section 8 (Confidentiality)) limits Khosla Tech's right to independently use, develop, evaluate, or market products, whether incorporating Feedback or otherwise.</p>

    <h2>8. Confidentiality</h2>
    <p>Except as otherwise set forth in Section 7 (Khosla Tech's Ownership Rights) above, each party agrees that all code, inventions, know-how, business, technical and financial information it obtains (\\"Receiving Party\\") from the disclosing party (\\"Disclosing Party\\") constitute the confidential property of the Disclosing Party (\\"Confidential Information\\"), provided that it is identified as confidential at the time of disclosure or should be reasonably known by the Receiving Party to be Confidential Information due to the nature of the information disclosed and the circumstances surrounding the disclosure. Any Khosla Tech Technology, performance information relating to the Products, and the terms and conditions of this Agreement shall be deemed Confidential Information of Khosla Tech without any marking or further designation. Except as expressly authorized herein, the Receiving Party will hold in confidence and not use or disclose any Confidential Information. The Receiving Party's nondisclosure obligation shall not apply to information which the Receiving Party can document: (i) was rightfully in its possession or known to it prior to receipt of the Confidential Information; (ii) is or has become public knowledge through no fault of the Receiving Party; (iii) is rightfully obtained by the Receiving Party from a third party without breach of any confidentiality obligation; (iv) is independently developed by employees of the Receiving Party who had no access to such information; or (v) is required to be disclosed pursuant to a regulation, law or court order (but only to the minimum extent required to comply with such regulation or order and with advance notice to the Disclosing Party). The Receiving Party acknowledges that disclosure of Confidential Information would cause substantial harm for which damages alone would not be a sufficient remedy, and therefore that upon any such disclosure by the Receiving Party the Disclosing Party shall be entitled to appropriate equitable relief in addition to whatever other remedies it might have at law.</p>

    <h2>9. Termination</h2>
    <p>This Agreement will continue so long as End User has a license or a subscription to the Software and the Hosted Services, unless earlier terminated. Khosla Tech may suspend or terminate this Agreement and End User's account, with respect to one or more of the Products, if End User fails to comply with the terms and conditions of this Agreement, including any failure to pay fees when due. Khosla Tech may terminate any free account or evaluation usage at any time in its sole discretion.</p>
    <p>End User may terminate this Agreement at any time with notice to Khosla Tech.</p>
    <p>Immediately upon termination of any license or subscription right granted under this Agreement, End User's license to Software and Hosted Services will cease, and End User must at its own cost: (a) cease using (and require all Authorized Users, and anyone else to cease using) all the terminated Products; (b) remove all copies of Software from its computer systems and any Uncontrolled Systems; and (c) return to Khosla Tech all Software or provide Khosla Tech with written certification that it has destroyed all copies of the Software and other Khosla Tech Confidential Information in its possession, custody or control. Upon termination of this Agreement for whatever reason, End User will not be entitled to credits or refunds for any unused portion of this Agreement, including but not limited to unused maintenance and support.</p>
    <p>All payment and other obligations accrued as of the date of any expiration or termination of this Agreement, and Sections 4.4 (Conditions on Use of Products), 4.5 (Indemnification), 5 (Fees and Payment), 6 (License Certifications and Audit), 7 (Khosla Tech's Ownership Rights), 8 (Confidentiality), 9 (Termination), 10 (Warranties; Disclaimer of Warranties), 11 (Limitation of Liability), 12 (Publicity Rights), 13 (Assignment), 14 (Governing Law and Arbitration), and 15 (General) will survive expiration or termination.</p>

    <h2>10. Disclaimers of Warranties</h2>
    <p><strong>10.1 General.</strong> Each party represents and warrants that it has the legal power and authority to enter into this Agreement, and that, if End User is an entity, this Agreement and each order is entered into by an employee or agent of such party with all necessary authority to bind such party to the terms and conditions of this Agreement.</p>
    <p><strong>10.2 DISCLAIMER OF WARRANTIES.</strong> TO THE MAXIMUM EXTENT PERMITTED BY APPLICABLE LAW, AND EXCEPT AS EXPRESSLY SET FORTH HEREIN, KHOSLA TECH AND ITS THIRD PARTY SUPPLIERS/LICENSORS PROVIDE THE PRODUCTS (INCLUDING THE HOSTED SERVICES, APIs, SOFTWARE AND ANY MAINTENANCE) AS IS AND WITH ALL FAULTS, AND HEREBY DISCLAIM ALL OTHER REPRESENTATIONS, WARRANTIES AND GUARANTEES, WHETHER EXPRESS, IMPLIED OR STATUTORY, INCLUDING IMPLIED WARRANTIES OF MERCHANTABILITY, TITLE, NON-INFRINGEMENT AND FITNESS FOR ANY PURPOSE. WITHOUT LIMITING THE FOREGOING, KHOSLA TECH AND ITS THIRD PARTY SUPPLIERS/LICENSORS MAKE NO REPRESENTATION, WARRANTY OR GUARANTY (1) AS TO THE RELIABILITY, TIMELINESS, QUALITY, SUITABILITY, TRUTH, AVAILABILITY, ACCURACY OR COMPLETENESS OF ANY PRODUCTS OR ANY CONTENT THEREIN OR GENERATED THEREWITH, (2) THAT (A) THE USE OF ANY PRODUCTS WILL BE SECURE, TIMELY, UNINTERRUPTED OR ERROR-FREE OR OPERATE IN COMBINATION WITH ANY OTHER HARDWARE, SOFTWARE, SYSTEM OR DATA, (B) THE PRODUCTS WILL MEET END USER'S REQUIREMENTS OR EXPECTATIONS, (C) ANY STORED DATA WILL BE ACCURATE OR RELIABLE OR THAT ANY STORED DATA WILL NOT BE LOST OR CORRUPTED, (D) THE QUALITY OF ANY PRODUCTS, SERVICES, INFORMATION, OR OTHER MATERIAL PURCHASED OR OBTAINED BY END USER THROUGH THE PRODUCTS WILL MEET END USER'S REQUIREMENTS OR EXPECTATIONS, (E) ERRORS OR DEFECTS WILL BE CORRECTED, OR (F) THE PRODUCTS (OR ANY SERVER(S) THAT MAKE A HOSTED SERVICE AVAILABLE) ARE FREE OF VIRUSES OR OTHER HARMFUL COMPONENTS. WITHOUT LIMITING THE FOREGIONG, TO THE MAXIMUM EXTENT PERMITTED BY APPLICBLE LAW, KHOSLA TECH AND ITS THIRD PARTY SUPPLIERS/LICENSORS MAKE NO REPRESENTATIONS OR WARRANTIES WHATSOEVER WITH RESPECT TO, AND ASSUME NO LIABILITY FOR, ANY PRODUCTS PROVIDED ON AN EVALUATION BASIS. IN ADDITION, KHOSLA TECH AND ITS THIRD PARTY LICENSOR/SUPPLIERS SHALL NOT BE LIABLE FOR DELAYS, INTERRUPTIONS, SERVICE FAILURES AND OTHER PROBLEMS INHERENT IN USE OF THE INTERNET AND ELECTRONIC COMMUNICATIONS OR OTHER SYSTEMS OUTSIDE THE REASONABLE CONTROL OF KHOSLA TECH.</p>
    <p>END USER MAY HAVE OTHER STATUTORY RIGHTS. HOWEVER, TO THE FULL EXTENT PERMITTED BY LAW, THE DURATION OF STATUTORILY REQUIRED WARRANTIES, IF ANY, SHALL BE LIMITED TO THE LIMITED WARRANTY PERIOD.</p>

    <h2>11. Limitation of Liability</h2>
    <p><strong>11.1 MAXIMUM LIABILITY.</strong> EXCEPT WITH RESPECT TO AMOUNTS PAYABLE TO THIRD PARTIES PURSUANT TO THE PARTIES' INDEMNIFICATION OBLIGATIONS IN SECTION 4.5 OR END USER'S BREACH OF ANY LICENSE OR USE RESTRICTIONS RELATING TO THE PRODUCTS, NOTWITHSTANDING ANY DAMAGES EITHER PARTY MIGHT INCUR FOR ANY REASON WHATSOEVER, TO THE MAXIMUM EXTENT PERMITTED BY APPLICABLE LAW, THE ENTIRE LIABILITY OF EITHER PARTY TO THIS AGREEMENT UNDER ANY PROVISION OF THIS AGREEMENT AND THE OTHER PARTY'S EXCLUSIVE REMEDY HEREUNDER SHALL BE LIMITED TO THE ACTUAL DAMAGES SUCH PARTY INCURS, UP TO THE AMOUNT ACTUALLY PAID BY END USER FOR (A) SUCH SOFTWARE, DEPRECIATED ON A THREE-YEAR STRAIGHT LINE BASIS, OR (B) SUCH HOSTED SERVICE IN THE LAST TWELVE (12) MONTHS PRECEDING THE DATE ON WHICH THE CLAIM AROSE.</p>
    <p><strong>11.2 OTHER DISCLAIMERS.</strong> NOTWITHSTANDING ANYTHING ELSE IN THIS AGREEMENT, TO THE MAXIMUM EXTENT PERMITTED BY APPLICABLE LAW, AND EXCEPT FOR END USER'S BREACH OF ANY LICENSE OR USE RESTRICTIONS RELATING TO PRODUCTS, IN NO EVENT SHALL EITHER PARTY (OR THEIR RESPECTIVE THIRD PARTY SUPPLIERS/LICENSORS) BE LIABLE FOR ANY SPECIAL, INCIDENTAL, PUNITIVE, INDIRECT, OR CONSEQUENTIAL DAMAGES (INCLUDING LOST PROFITS OR REVENUE) WHATSOEVER ARISING OUT OF OR IN ANY WAY RELATED TO THIS AGREEMENT. END USER SPECIFICALLY UNDERSTANDS AND AGREES THAT KHOSLA TECH (ON BEHALF OF ITSELF AND ITS THIRD PARTY SUPPLIERS/LICENSORS) DISCLAIMS ALL WARRANTIES AND LIABILITY WITH RESPECT TO LOSS, LOSS OF USE OR CORRUPTION OF ANY END USER DATA (OR OTHER DATA END USER MAY PROVIDE) AND THE COSTS OF PROCUREMENT OF ANY SUBSTITUTE GOODS, any damages whatsoever arising out of or related to your use of the software, Hosted Services, APIs, the content and other information obtained directly/indirectly from Khosla Tech.</p>
    <p><strong>11.3 Failure of Essential Purpose.</strong> The parties agree that the limitations specified in this Section 11 will survive and apply even if any limited remedy specified in this Agreement is found to have failed of its essential purpose.</p>
    <p><strong>11.4 Force Majeure.</strong> Neither party shall be liable hereunder by reason of failure or delay in the performance of its obligations hereunder (except for the non-payment of money) on account of strikes, shortages, riots, insurrection, war, acts of terrorism, fires, flood, storm, explosions, earthquakes, acts of God, governmental action, labor conditions, or any other cause which is beyond the reasonable control of the party.</p>

    <h2>12. Publicity Rights</h2>
    <p>Khosla Tech may identify End User as a customer in Product promotional material. End User may request that Khosla Tech cease identifying End User at any time by submitting an email to sales@Khosla Tech.com. Requests may take 30 days to process.</p>

    <h2>13. Assignment</h2>
    <p>End User may not assign this Agreement without the prior written consent of Khosla Tech (which consent will not be unreasonably withheld), provided that the assignee agrees to be bound by the terms and conditions contained in this Agreement. Khosla Tech may assign its rights and obligations under this Agreement in whole or in part without consent of End User. Any permitted assignee shall be bound by the terms and conditions of this Agreement.</p>

    <h2>14. Governing Law and Arbitration</h2>
    <p>Any disputes arising out of or related to this Agreement and/or any use by you of the Khosla Tech’s Products shall be governed by the laws of India, without regard to the conflicts of laws provisions therein.</p>

    <h2>15. General</h2>
    <p>This Agreement is the entire agreement between End User and Khosla Tech relating to the Products and supersedes all prior or contemporaneous oral or written communications, proposals and representations with respect to the Products or any other subject matter covered by this Agreement. If any provision of this Agreement is held to be void, invalid, unenforceable or illegal, the other provisions shall continue in full force and effect. As used herein, \\"including\\" (and its variants) means \\"including without limitation\\" (and its variants). The parties are independent contractors. This Agreement shall not be construed as constituting either party as a partner of the other or to create any other form of legal association that would give on party the express or implied right, power or authority to create any duty or obligation of the other party.</p>
    <p>Should you have any questions concerning this Agreement, or if you desire to contact Khosla Tech for any reason, please <a href=\\"mailto:sales@ASPSecurityKit.net\\">contact us</a>.</p>
</div>
"
`;

exports[`Failed to fetch a license returns helper text instead 1`] = `"Did not fetch license text successfully."`;
